controllers/configurationpolicy_controller.go

```diff
@@ -1498,25 +1466,25 @@ func (r *ConfigurationPolicyReconciler) handleObjects(
 			"name", objName,
 			"kind", kindWithoutNS,
 		)
+
+		var space string
+		if objName != "" {
+			space = " "
+		}
+
 		// namespaced but none specified, generate violation
-		msg := fmt.Sprintf("namespaced object %s of kind %s has no namespace specified "+
+		msg := fmt.Sprintf("namespaced object%s%s of kind %s has no namespace specified "+
 			"from the policy namespaceSelector nor the object metadata",
-			objName, kindWithoutNS,
+			space, objName, kindWithoutNS,
 		)
-		statusUpdateNeeded = addConditionToStatus(policy, index, false, "K8s missing namespace", msg)
-
-		if statusUpdateNeeded {
-			eventType := eventNormal
-			if index < len(policy.Status.CompliancyDetails) &&
-				policy.Status.CompliancyDetails[index].ComplianceState == policyv1.NonCompliant {
-				eventType = eventWarning
-			}
 
-			r.Recorder.Event(policy, eventType, fmt.Sprintf(eventFmtStr, policy.GetName(), objDetails.name),
-				convertPolicyStatusToString(policy))
+		result = objectResult{
+			[]string{objName},
```

Comment:
It ends up not being used because it's an error case, so it falls in the `default` switch case. Here is an example of the status:
```yaml
      conditions:
        - lastTransitionTime: "2023-06-21T12:24:28Z"
          message: namespaced object of kind Role has no namespace specified from the policy namespaceSelector nor the object metadata
          reason: K8s missing namespace
          status: "True"
          type: violation
```

Reply:
I added a unit test for this situation to illustrate this. I hope this helps clarify it.